### PR TITLE
Render columns dynamically on wide tables

### PR DESCRIPTION
**Pull request by betodealmeida**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

With https://github.com/apache/incubator-superset/pull/7625 SQL Lab now displays complex column types as extended columns. This might result in very wide tables that degrade performance when visualizing preview samples or query results.

This PR modifies SQL Lab so that wide tables (greater than 50 columns) are displayed using `react-virtualized`'s `Grid` instead of a `Table`. This ensures that columns are loaded dynamically as the user scrolls horizontally. The only downside is that the user is no longer able to sort the data by clicking on a column header.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Here's a demo showing the `Grid` component with the `query` table (the table has only 27 columns, but I modified the threshold temporarily in order to test):

grid-view

I also tested with a table with over 300 columns, but I can't share the data.

### TEST PLAN

Tested locally (see above). Added unit test.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [X] Removes existing feature or API

This removes the ability of sorting on the client side on tables with more than 50 columns.

### REVIEWERS

@mistercrunch @khtruong

### CODE REVIEW

**Comment by etr2460** [comment]
The issue you're solving might be fixed by my PR here: https://github.com/apache/incubator-superset/pull/7690. If you pull that branch, do you still see the degraded performance?

File: superset/assets/src/components/FilterableTable/FilterableTable.jsx [hunk 1]
```diff
@@ -167,7 +179,95 @@ export default class FilterableTable extends PureComponent {
...
-  render() {
+  renderGridCellHeader({ columnIndex, key, style }) {
+    const label = this.props.orderedColumnKeys[columnIndex];
+    const className = this.props.expandedColumns.indexOf(label) > -1
+      ? 'header-style-disabled'
+      : 'header-style';
+    return (
+      <TooltipWrapper key={key} label="header" tooltip={label}>
+        <div
+          style={{ ...style, top: style.top - 4 }}
```
Comment (khtruong) [inline superset/assets/src/components/FilterableTable/FilterableTable.jsx]: Can we add a const above for 4.

File: superset/assets/src/components/FilterableTable/FilterableTable.jsx [hunk 2]
```diff
@@ -167,7 +179,95 @@ export default class FilterableTable extends PureComponent {
...
+            className="filterable-table-container Table"
+            ref={(ref) => { this.container = ref; }}
+          >
+            <div className="LeftColumn">
+              <Grid
+                cellRenderer={this.renderGridCellHeader}
+                columnCount={orderedColumnKeys.length}
+                columnWidth={getColumnWidth}
+                height={rowHeight}
+                ref={(ref) => { this.container = ref; }}
```
Comment (khtruong) [inline superset/assets/src/components/FilterableTable/FilterableTable.jsx]: Let's pull this out into a const so we aren't creating a new function every time. Comment applies elsewhere.

**Comment by betodealmeida** [comment]
@etr2460 that's awesome! I'll take a look.

I have a feeling that we might still need both solutions combined, since this one didn't solve all the performance problems. And our table when rendered has more than 2 millions pixels of width, so your solution might not be enough.

File: superset/assets/src/components/FilterableTable/FilterableTable.jsx (hunk 1 above)
Comment (betodealmeida) [reply superset/assets/src/components/FilterableTable/FilterableTable.jsx]: Ah, good catch! I'll fix it.

File: superset/assets/src/components/FilterableTable/FilterableTable.jsx (hunk 2 above)
Comment (betodealmeida) [reply superset/assets/src/components/FilterableTable/FilterableTable.jsx]: +1

**Review by khtruong — APPROVED** [review]

**Pull request merged**